akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/receptionist/ClusterReceptionist.scala

```diff
@@ -253,6 +261,27 @@ private[typed] object ClusterReceptionist extends ReceptionistBehaviorProvider {
           }
           Behaviors.same
 
+        case ReceptionistMessages.Unregister(key, serviceInstance, maybeReplyTo) =>
+          val entry = Entry(serviceInstance, setup.selfSystemUid)
+          ctx.log.debugN(
+            "ClusterReceptionist [{}] - Unregister actor: [{}] [{}]",
+            cluster.selfAddress,
+            key.asServiceKey.id,
+            entry)
+          val watcher = terminationWatchers.remove(uniqueKey(key, serviceInstance))
+          if(!watcher.isEmpty) ctx.stop(watcher.get)
+          maybeReplyTo match {
+            case Some(replyTo) => replyTo ! ReceptionistMessages.Unregistered(key, serviceInstance)
+            case None          =>
+          }
+          val ddataKey = registry.ddataKeyFor(key)
+          replicator ! Replicator.Update(ddataKey, EmptyORMultiMap, settings.writeConsistency) { registry =>
+            ServiceRegistry(registry).removeBinding(key, entry).toORMultiMap
+          }
+          // tombstone removals so they are not re-added by merging with other concurrent
+          // registrations for the same key
+          next(newState = registry.addTombstone(serviceInstance, setup.newTombstoneDeadline()))
```

Comment:
I think there may be some new scenarios now, because with actor stopping as only reason for deregistration we would never get the same actor re-added. We may need to reconsider the tombstone with re-registering in mind. 

We definitely will need added test coverage for this.

Reply:
That is true, I remember I had a comment about that back in https://github.com/akka/akka/pull/25699#discussion_r222586829